erpnext/stock/doctype/delivery_trip/delivery_trip.py

```diff
@@ -47,6 +48,17 @@ def validate_stop_addresses(self):
 			if not stop.customer_address:
 				stop.customer_address = get_address_display(frappe.get_doc("Address", stop.address).as_dict())
 
+	def is_visited(self):
+		for stop in self.delivery_stops:
+			if stop.visited:
+				status = frappe.db.get_value("Sales Invoice", stop.sales_invoice, "status")
+				if status == "Unpaid":
+					frappe.db.set_value("Delivery Note", stop.delivery_note, "status", "Delivered")
+				if status == "Paid":
+					frappe.db.set_value("Delivery Note", stop.delivery_note, "status", "Completed")
+			else:
+				frappe.db.set_value("Delivery Note", stop.delivery_note, "status", "To Deliver")
```

Comment:
If this is happening in the delivery trip, I think you should also check if the trip has been started, cause it can be either "To Deliver" or "Out for Delivery"